packages/amplify-graphql-relational-transformer/src/resolvers.ts

```diff
@@ -8,6 +8,57 @@ import { getSortKeyFields } from './schema';
 import { HasManyDirectiveConfiguration } from './types';
 import { getConnectionAttributeName, getObjectPrimaryKey } from './utils';
 
+export const updateTableForReferencesConnection = (
```

Comment:
Lots of notes through this method. I think my confusion stems from my assumptions about the configuration of the `hasMany` directive coming into this method. Right now it's opaque what each of the members means.

Given a schema like

```graphql
type Primary @model { 
  content: String
  related: Related @hasMany(references: 'primaryId')
}
type Related @model {
  content: String
  primaryId: ID!
  primary: Primary @belongsTo(references: 'primaryId')
}
```

...my assumptions (which I partially validated by reading through related code, but can't claim they're exhaustive) are:

- `field` - the GraphQL FieldDefinitionNode on which the `hasMany` directive is declared (i.e., `Primary.related`)
- `fieldNodes` - the GraphQL FieldDefinitionNodes for the fields referenced by the `fields` attribute (invalid for references-style relationships)
- `indexName` - the customer-specified index name to use for the relationship (invalid for references-style relationships)
- `object` - the GraphQL ObjectTypeDefinitionNode of the **parent object** of the field on which the `hasMany` directive is declared (i.e., `Primary`)
- `references` - a list of strings: the value of the `references` attribute
- `relatedType` - the GraphQL ObjectTypeDefinitionNode of the type of the field on which the `hasMany` directive is declared (i.e., `Related`)

Further, I'm assuming the intent of this method is to create a GSI on the related table so we can resolve hasMany and belongsTo relationships by querying that GSI.

If those are accurate, then I don't follow that the logic in the method is creating a properly configured GSI on the related table.